### PR TITLE
Moving more files to Common

### PR DESCRIPTION
**Pull request by penev92**
- Moved `GivesBounty`, `GivesExperience` and `GainsExperience` traits, along with the `Rank` effect
- Moved the `DetectCloaked` trait
- Moved the `PlayerStatistics` trait
- Moved the `Enter` and `RepairBuilding` activities
- Moved the `EngineerRepair` trait
- Moved `ActorExts`
- Extracted the `IgnoresDisguise` trait to its own file

### CODE REVIEW

File: OpenRA.Mods.Common/OpenRA.Mods.Common.csproj [hunk 1]
```diff
@@ -284,4 +295,4 @@ cd "$(SolutionDir)"</PostBuildEvent>
   <Target Name="AfterBuild">
   </Target>
   -->
-</Project>
+</Project>
```
Comment (reaperrr) [inline OpenRA.Mods.Common/OpenRA.Mods.Common.csproj]: I faintly remember this was a topic in the past, but I don't remember whether we wanted to have a newline at the end of .csproj or not. Just pointing it out so it doesn't get overlooked by other reviewers.

File: OpenRA.Mods.RA/OpenRA.Mods.RA.csproj [hunk 2]
```diff
@@ -444,4 +434,4 @@ copy "FuzzyLogicLibrary.dll" "$(SolutionDir)"
 cd "$(SolutionDir)"</PostBuildEvent>
   </PropertyGroup>
   <ItemGroup />
-</Project>
+</Project>
```
Comment (reaperrr) [inline OpenRA.Mods.RA/OpenRA.Mods.RA.csproj]: Same here.

**Comment by Phrohdoh** [comment]
Personal nit: Commit messages should say `Move Give/Gains Experiance and Bounty to Mods.Common` instead of just `Moves abc`.

Also, `Experiance` is misspelled.

File: OpenRA.Mods.Common/OpenRA.Mods.Common.csproj (hunk 1 above)
Comment (penev92) [inline OpenRA.Mods.Common/OpenRA.Mods.Common.csproj]: I see absolutely no reason to have it.

**Comment by penev92** [comment]
Renamed the commits.

File: OpenRA.Mods.Common/OpenRA.Mods.Common.csproj (hunk 1 above)
Comment (Mailaender) [inline OpenRA.Mods.Common/OpenRA.Mods.Common.csproj]: Nobody cares. Just don't make any unnecessary changes to these files.

File: OpenRA.Mods.RA/OpenRA.Mods.RA.csproj (hunk 2 above)
Comment (Mailaender) [inline OpenRA.Mods.RA/OpenRA.Mods.RA.csproj]: Yes, please don't commit these small out of context changes the IDE does.

File: OpenRA.Mods.Common/OpenRA.Mods.Common.csproj (hunk 1 above)
Comment (penev92) [inline OpenRA.Mods.Common/OpenRA.Mods.Common.csproj]: Sure. I'll tell my IDE and git client that they've been bad and shouldn't try to fix stuff someone screwed up long ago, because that's how things are done around here.

**Comment by Mailaender** [comment]
:+1: on the file and namespace movements.

**Comment by Phrohdoh** [comment]
:+1:
